collection-scripts/gather_services_status

```diff
@@ -77,6 +82,28 @@ get_cinder_status() {
     run_bg ${BASH_ALIASES[os]} volume backend pool list --long '>' "$CINDER_PATH"/pool_list
 }
 
+# Nova service gathering - sevices, hypervisors, cells, host mappings, allocation audit
+get_nova_status() {
+    local NOVA_PATH="$BASE_COLLECTION_PATH/ctlplane/nova"
+    mkdir -p "$NOVA_PATH"
+    run_bg ${BASH_ALIASES[os]} compute service list '>' "$NOVA_PATH"/service_list
+    run_bg ${BASH_ALIASES[os]} hypervisor list '>' "$NOVA_PATH"/hypervisor_list
+    run_bg /usr/bin/oc -n openstack exec -t nova-cell0-conductor-0 -- nova-manage cell_v2 list_cells '>' "$NOVA_PATH"/cell_list
+    run_bg /usr/bin/oc -n openstack exec -t nova-cell0-conductor-0 -- nova-manage cell_v2 list_hosts '>' "$NOVA_PATH"/host_list
+    run_bg /usr/bin/oc -n openstack exec -t nova-cell0-conductor-0 -- nova-manage placement audit --verbose '>' "$NOVA_PATH"/nova-manage_placement_audit
+    run_bg /usr/bin/oc -n openstack exec -t nova-cell0-conductor-0 -- nova-manage placement heal_allocations --dry-run --verbose '>' "$NOVA_PATH"/nova-manage_placement_heal_allocations
```

Comment:
I slept on it and these might be too expensive operations for a large cloud as both examine each instances.
@SeanMooney  what do you think?

Reply:
the heal allocation and auit command scale based on the number of allocations (instance + migratuions) so on a large cloud this could be minutes

i would not inc them in the nova section personally at least not by default.

can we parmatierise this? i.e. only include it if its really needed.
i think I'm leaning more towards saying if we need this info then we should ask them to run those command separately since in general i don't think we will used that output to debug a customer issue often.

Reply:
by the way i knwo these are nova commands but this feels more like it should be in placement. in any case i would not consider the instance allocation to be an attribute of the nova status in general that's an attribute of the workloads.

if we are considering "workload" type data in the status
there are a number of other things we could consider but it would require API access

flavor list with extra specs per flavour.
aggregate list, aggrate metadata show and aggregate host list.

cinder is including volume type list so that is basically the same as our flavors
neutorn is basically dumping all the workload object (port,networks, ectra)

so i think including the aggregate info is in scope. that would allow use to diagnose most schuilign issues (if we also have image properties and the placement summaries you are collecting below)

Reply:
> the heal allocation and auit command scale based on the number of allocations (instance + migratuions) so on a large cloud this could be minutes
> 
> i would not inc them in the nova section personally at least not by default.
> 
> can we parmatierise this? i.e. only include it if its really needed. i think I'm leaning more towards saying if we need this info then we should ask them to run those command separately since in general i don't think we will used that output to debug a customer issue often.

If you both think this can take a while and it won't help in most cases we can definitely do it only upon request.

The way we are currently parametrizing things is using environmental variables, so that could be used to have a flag to gather extra stuff, and then in our docs we could define what cases that should be gathered.

Reply:
I will remove it as we agree it takes too much time to do in each must gather. The parameterization can be done separately. But if we need to ask a specific thing to add as a parameter that is very similar effort to actually send to command to run to gather the extra output. So I will not add parameterization now.

Reply:
@SeanMooney we have API access, so we can list flavors. But I'm not sure that is useful in general. Also that can be slow e.g. try to list flavors in PSI :)